cogeo_mosaic/backends/base.py

```diff
@@ -38,50 +38,60 @@ class BaseBackend(BaseReader):
     Attributes:
         input (str): mosaic path.
         mosaic_def (MosaicJSON, optional): mosaicJSON document.
+        tms (morecantile.TileMatrixSet, optional): TileMatrixSet grid definition. Defaults to `WebMercatorQuad`.
+        minzoom (int): mosaic Min zoom level. Defaults to tms or mosaic minzoom.
+        maxzoom (int): mosaic Max zoom level. Defaults to tms or mosaic maxzoom.
         reader (rio_tiler.io.BaseReader): Dataset reader. Defaults to `rio_tiler.io.Reader`.
         reader_options (dict): Options to forward to the reader config.
         geographic_crs (rasterio.crs.CRS, optional): CRS to use as geographic coordinate system. Defaults to WGS84.
-        tms (morecantile.TileMatrixSet, optional): TileMatrixSet grid definition. **READ ONLY attribute**. Defaults to `WebMercatorQuad`.
-        bbox (tuple): mosaic bounds (left, bottom, right, top). **READ ONLY attribute**. Defaults to `(-180, -90, 180, 90)`.
-        minzoom (int): mosaic Min zoom level. **READ ONLY attribute**. Defaults to `0`.
-        maxzoom (int): mosaic Max zoom level. **READ ONLY attribute**. Defaults to `30`
+        bounds (tuple): mosaic bounds (left, bottom, right, top). **READ ONLY attribute**. Defaults to `(-180, -90, 180, 90)`.
+        crs (rasterio.crs.CRS): mosaic crs (left, bottom, right, top). **READ ONLY attribute**. Defaults to `(-180, -90, 180, 90)`.
 
     """
 
     input: str = attr.ib()
     mosaic_def: MosaicJSON = attr.ib(default=None, converter=_convert_to_mosaicjson)
 
+    tms: TileMatrixSet = attr.ib(default=WEB_MERCATOR_TMS)
+    minzoom: int = attr.ib(default=None)
+    maxzoom: int = attr.ib(default=None)
+
     reader: Union[
         Type[BaseReader],
         Type[MultiBaseReader],
         Type[MultiBandReader],
     ] = attr.ib(default=Reader)
     reader_options: Dict = attr.ib(factory=dict)
 
-    geographic_crs: CRS = attr.ib(default=WGS84_CRS)
-
-    # TMS is outside the init because mosaicJSON and cogeo-mosaic only
-    # works with WebMercator for now.
-    tms: TileMatrixSet = attr.ib(init=False, default=WEB_MERCATOR_TMS)
-    minzoom: int = attr.ib(init=False)
-    maxzoom: int = attr.ib(init=False)
-
-    # default values for bounds
     bounds: Tuple[float, float, float, float] = attr.ib(
         init=False, default=(-180, -90, 180, 90)
     )
     crs: CRS = attr.ib(init=False, default=WGS84_CRS)
+    geographic_crs: CRS = attr.ib(init=False, default=WGS84_CRS)
```

Comment:
there is a number of places where the crs and geographic_crs are defined, but shouldn't the geographic_crs always correspond to the one used by the crs (projection or otherwise)? So it would be easier to just only ever explicitly define the crs and always do `crs.geographic_crs` unless I am missing something?

Reply:
> but shouldn't the geographic_crs always correspond to the one used by the crs (projection or otherwise)? 

Sadly in Rasterio's CRS the `geodetic_crs` property is not available which is why we have separate property in rio-tiler (https://github.com/cogeotiff/rio-tiler/blob/main/rio_tiler/io/base.py#L42-L44)

Reply:
ahh forgot it wasn't a Proj CRS